test/__snapshots__/transform.test.ts.snap

```diff
@@ -330,6 +330,72 @@ export default __sfc_main;
 "
 `;
 
+exports[`transform > fixtures > test/fixtures/HtmlTag.vue 1`] = `
+"<script lang=\\"ts\\">
+import Enum from './Enum.vue';
+import { ref } from '@vue/composition-api';
+const __sfc_main = {};
+__sfc_main.setup = (__props, __ctx) => {
+  let p = \\"hello word\\";
+  let Div = ref(\\"hello word\\");
+  let h3 = 'test';
+  let H3 = '';
+  return {
+    p,
+    Div,
+    h3,
+    H3
+  };
+};
+__sfc_main.components = Object.assign({
+  Enum
+}, __sfc_main.components);
+export default __sfc_main;
+</script>
+
+<template>
+  <div>
+    <Enum/>
+    <h3></h3>
+    {{ H3 }}
+    {{ h3 }}
+    {{ Div }}
+    <p>{{ p }}</p>
+  </div>
+</template>
+"
+`;
+
+exports[`transform > fixtures > test/fixtures/HtmlTag2.vue 1`] = `
+"<script lang=\\"ts\\">
+import Button from './DynamicStyle.vue';
+import button from './DynamicStyle.vue';
+import { defineAsyncComponent } from '@vue/composition-api';
+const footer = defineAsyncComponent(() => import('./ScriptOnly.vue'));
+import { ref } from '@vue/composition-api';
+const __sfc_main = {};
```

Comment:
This seems wrong in vue 3.
We should add `__sfc_main.components = {Button}`

see example:
https://play.vuejs.org/#eNqNUcFqwzAM/RXhSxMY9b1zAt3YroWxoy9d4mwBxza20jFC/n1y7aYttNBTLL0n6b2XiW2dWx9GxTZMhMb3DiEoHB3ovfmuJMMgWS1NPzjrEV5GRGug83aA1ZqnMo6vnhfO1wOcCVrV9UZtw59pXi01jTIIc56KZGkaawJCZy0qD9XNiaIooaohbS2ur5XlxTmvuqvtJCWtd1ARVkj2o7S28Gt9K1kJhAue8iD3VKAanN6jogpAtP3h+KCnq6eJtsyz4O7USxFEIAcW0dzMlARESs4rUnIzU5Lz+n23+3z7EDyXx/s8CRB8kcWe2Nn77Z+JIVk5+7rrKstect6ACNpiTbPxc0/D/A8NBMz/

Reply:
ok finished
![Snipaste_2023-06-19_14-06-42](https://github.com/antfu/unplugin-vue2-script-setup/assets/53564781/25fa3235-e478-46cd-b501-a2c96c0a178e)